src/lib/device/physical_dev.cpp

```diff
@@ -220,8 +220,10 @@ void PhysicalDev::submit_batch() { m_drive_iface->submit_batch(); }
 
 //////////////////////////// Chunk Creation/Load related methods /////////////////////////////////////////
 void PhysicalDev::format_chunks() {
-    m_chunk_info_slots = std::make_unique< sisl::Bitset >(hs_super_blk::chunk_info_bitmap_size(m_dev_info));
+    m_chunk_info_slots = std::make_unique< sisl::Bitset >(std::max(1u, hs_super_blk::max_chunks_in_pdev(m_dev_info)));
     auto bitmap_mem = m_chunk_info_slots->serialize(m_pdev_info.dev_attr.align_size);
```

Comment:
We are asking the bitset library to serialize based on pdev align size. So I don't see anything wrong here.

Reply:
in `chunk_info_bitmap_size` we round up to 4K instead of pdev align size.  see the example here https://github.com/eBay/HomeStore/pull/415#discussion_r1601120888 we might hit the assert.